File: sdk_ceshi_demo/ShenZhouDC-OC-iPhone/shenzhoudc-iPhone/我的/VC/AboutUsViewController.m
//
//  AboutUsViewController.m
//  shenzhoudc-iPhone
//
//  Created by 张丹丹 on 16/12/30.
//  Copyright © 2016年 Eteclabeteclab. All rights reserved.
//

#import "AboutUsViewController.h"
#import "CommentViewController.h"

@interface AboutUsViewController ()

@end

@implementation AboutUsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"关于我们";
    self.view.backgroundColor = [UIColor whiteColor];
    [self makeUI];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void)makeUI {
    
    UIImageView *logo = [[UIImageView alloc]init];
    logo.image = [UIImage imageNamed:@"logoyun"];
    [self.view addSubview:logo];
    [logo mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerX.mas_equalTo(self.view.mas_centerX);
        make.top.mas_equalTo(SCREEN_WIDTH/6+LandscapeNumber(64));
        make.size.mas_equalTo(CGSizeMake(SCREEN_WIDTH*16/75, SCREEN_WIDTH*16/75));
    }];
    
    UILabel *titleLabel = [[UILabel alloc]init];
    titleLabel.text = @"神州方案云";
    titleLabel.font = [UIFont systemFontOfSize:20];
    titleLabel.textColor = UIColorFromRGB(0xD71629);
    [self.view addSubview:titleLabel];
    [titleLabel mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerX.mas_equalTo(self.view.mas_centerX);
        make.top.equalTo(logo.mas_bottom).offset(LandscapeNumber(15));
    }];
    
    UILabel *verson = [[UILabel alloc]init];
    verson.text = @"v 1.0";
    verson.font = [UIFont systemFontOfSize:16];
    verson.textColor = UIColorFromRGB(0xD71629);
    [self.view addSubview:verson];
    [verson mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerX.mas_equalTo(self.view.mas_centerX);
        make.top.equalTo(titleLabel.mas_bottom).offset(LandscapeNumber(15));
    }];
    
    
    
//    UIButton *btn = [[UIButton alloc]init];
//    btn.backgroundColor = [UIColor yellowColor];
//    [btn addTarget:self action:@selector(cliclbtyn) forControlEvents:UIControlEventTouchUpInside];
//    [self.view addSubview:btn];
//    [btn mas_makeConstraints:^(MASConstraintMaker *make) {
//        make.centerX.mas_equalTo(self.view.mas_centerX);
//        make.top.equalTo(verson.mas_bottom).offset(LandscapeNumber(20));
//        make.size.mas_equalTo(CGSizeMake(50, 50));
//    }];
    
    UILabel *label = [[UILabel alloc]init];
    label.text = @"Copyright ©2017 神州方案云 版权所有";
    label.numberOfLines = 0;
    label.textAlignment = NSTextAlignmentCenter;
    label.font = [UIFont systemFontOfSize:12];
    label.textColor = UIColorFromRGB(0x999999);
    [self.view addSubview:label];
    [label mas_makeConstraints:^(MASConstraintMaker *make) {
        make.centerX.mas_equalTo(self.view.mas_centerX);
        make.bottom.mas_equalTo(-LandscapeNumber(16));
        make.width.mas_equalTo(LandscapeNumber(249));
    }];
    
    UIView *line = [[UIView alloc]init];
    line.backgroundColor = [UIColor blackColor];
    [self.view addSubview:line];
    [line mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.mas_equalTo(10);
        make.bottom.equalTo(label.mas_top).offset(-LandscapeNumber(16));
        make.size.mas_equalTo(CGSizeMake(SCREEN_WIDTH-20, 1));
    }];
    
    
    
}
- (void)cliclbtyn {
    CommentViewController *vc = [[CommentViewController alloc]init];
    [self.navigationController pushViewController:vc animated:YES];
}

@end

File: sdk_ceshi_demo/NebulaCommunitySwift/WashingMachine/Me/Controller/AboutUSViewController.swift
//
//  AboutUSViewController.swift
//  WashingMachine
//
//  Created by 张丹丹 on 16/11/2.
//  Copyright © 2016年 Eteclabeteclab. All rights reserved.
//

import UIKit
import SnapKit

class AboutUSViewController: BaseViewController {

    fileprivate var verson:UILabel?
    fileprivate var contentTextView: UITextView!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        self.automaticallyAdjustsScrollViewInsets = false
        self.navigationItem.title = "关于我们"
        self.view.backgroundColor = UIColor.white
        
        self.makeUI()
        self.aboutUS()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    fileprivate func aboutUS() {
        let url = SERVICE_BASE_ADDRESS + API_GET_APP_ABOUT
        NetworkEngine.get(url, parameters: nil) { (result) in
            if result.isSuccess {
                if let dict = result.dataObj as? [String: AnyObject] {
                    let data = dict["data"] as? String ?? ""
                    self.contentTextView.text = data
                }
            }
        }
    }

    fileprivate func makeUI() {
        
        let logo = UIImageView()
        logo.image = UIImage(named: "logo")
        self.view.addSubview(logo)
        logo.snp.makeConstraints { (make) in
            make.centerX.equalToSuperview()
            make.top.equalToSuperview().offset(24)
            make.height.width.equalTo(BOUNDS_WIDTH*4/15)
        }
        
        let label = UILabel()
        label.text = "Copyright ©2016 北京时代星云科技有限公司 版权所有"
        label.numberOfLines = 0
        label.textAlignment = .center
        label.font = UIFont.systemFont(ofSize: 12)
        label.textColor = UIColor.init(red: 102/255.0, green: 102/255.0, blue: 102/255.0, alpha: 1)
        self.view.addSubview(label)
        label.snp.makeConstraints { (make) in
            make.centerX.equalToSuperview()
            make.bottom.equalToSuperview().offset(-16)
            make.left.equalToSuperview().offset(35)
            make.right.equalToSuperview().offset(-35)
        }
        
        let line = UIView()
        line.backgroundColor = UIColor.black
        self.view.addSubview(line)
        line.snp.makeConstraints { (make) in
            make.left.equalToSuperview().offset(10)
            make.right.equalToSuperview().offset(-10)
            make.bottom.equalTo(label.snp.top).offset(-16)
            make.height.equalTo(0.5)
        }
        
        
        verson = UILabel()
        verson?.text = "版本号  " + HZApp.appVersion
        verson?.font = UIFont.systemFont(ofSize: 16)
        verson?.textColor = THEMEGRAYCOLOR
        self.view.addSubview(verson!)
        verson?.snp.makeConstraints({ (make) in
            make.centerX.equalToSuperview()
            make.bottom.equalTo(line.snp.top).offset(-24);
        })
     
        let centerV = UIView()
        centerV.backgroundColor = UIColor(rgb:0xF3F8FB)
        centerV.layer.masksToBounds = true
        centerV.layer.cornerRadius = 4
        self.view.addSubview(centerV)
        centerV.snp.makeConstraints { (make) in
            make.left.equalToSuperview().offset(12)
            make.right.equalToSuperview().offset(-12)
            make.top.equalTo(logo.snp.bottom).offset(24)
            make.bottom.equalTo((self.verson?.snp.top)!).offset(-37)
        }
     
        let titleLabel = UITextView()
        titleLabel.font = UIFont.systemFont(ofSize: 14)
        titleLabel.isEditable = false
        titleLabel.textColor = UIColor.init(red: 51/255.0, green: 51/255.0, blue: 51/255.0, alpha: 1)
        centerV.addSubview(titleLabel)
        self.contentTextView = titleLabel
        titleLabel.snp.makeConstraints { (make) in
            make.left.top.equalToSuperview().offset(10)
            make.right.bottom.equalToSuperview().offset(-10)
        }
    }
}
